src/Storages/MergeTree/MergeTreePrefetchedReadPool.cpp

```diff
@@ -514,22 +519,59 @@ MergeTreePrefetchedReadPool::ThreadsTasks MergeTreePrefetchedReadPool::createThr
 
             read_task->priority = priority;
 
-            bool allow_prefetch = !settings.filesystem_prefetches_limit || current_prefetches_count + 1 <= settings.filesystem_prefetches_limit;
-            if (allow_prefetch && settings.filesystem_prefetch_max_memory_usage)
+            bool allow_prefetch = false;
+            if (allowed_memory_usage
+                && (allowed_prefetches_num.has_value() == false || allowed_prefetches_num.value() > 0))
```

Comment:
this condition is always true

Reply:
this is not true, it could become because of these lines https://github.com/ClickHouse/ClickHouse/blob/88784568dcf8f32892fcdd5ffefe28320c279a76/src/Storages/MergeTree/MergeTreePrefetchedReadPool.cpp#L567-L569

Reply:
indeed, my mistake